### PR TITLE
Load all enabled apps in test bootstrap

### PR DESCRIPTION
**Pull request by RobinMcCorkell**
A possible solution to the chaos caused by #18839. We should load all apps that have been enabled, so that autoloader valid roots are properly registered (and so that all `app.php` is correctly run in case there's special initialization logic). Previously, we only loaded 'minimal' apps.

Reasoning:

In autotest.sh, we enable the apps we want to test. The same happens with autotest-external.sh. The autotest executor looks at all enabled apps and runs their tests, so if an app isn't loaded it will break at this point, as the paths are not valid.

When running tests manually for individual apps (aka outside of autotest), the dev must make sure that the app is enabled in the running instance before attempting to run tests against it, OR the app test bootstrap explicitly loads the app. I don't see any situations where it is desired that an app isn't loaded, but still has tests run against it.

Please state if this is an acceptable solution @nickvergessen @oparoz @ChristophWurst 

cc @MorrisJobke @LukasReschke

### CODE REVIEW

**Comment by Xenopathic** [comment]
TL;DR version:

For running app unit tests, either make sure you load core's `bootstrap.php` as per the recommended guidelines (and have the app enabled when you run your tests), or explicitly load your app in your `bootstrap.php` with the following line:

    \OC_App::loadApp('my_awesome_app');`

**Comment by Xenopathic** [comment]
To test: try running `./autotest-external.sh mysql common-tests`. Before this PR, it fails, now it succeeds

**Comment by oparoz** [comment]
I think it makes sense for the core bootstrap since that will test the system as it's delivered, with the installed apps made available in the tests.
For apps which require a custom boostrap, adding `\OC_App::loadApp` is simple enough.
:+1:

**Comment by LukasReschke** [comment]
:+1:
